src/components/PopoverProvider/index.tsx

```diff
@@ -93,6 +93,16 @@ function PopoverContextProvider(props: PopoverContextProps) {
         };
     }, [closePopover]);
 
+    React.useEffect(() => {
+        const listener = () => {
+            closePopover();
+        };
+        document.addEventListener('dragenter', listener);
+        return () => {
+            document.removeEventListener('dragenter', listener);
+        };
+    }, [closePopover]);
+
```

Comment:
Do we need this block in the context of the new call?

Reply:
Again, does dragging a file over LHN trigger the "ready-to-drop" behavior on `main`? I don't think it does. And it cannot cause the visual glitch.

If the drag gesture is "discarded" by the app (when we decide not to handle it), closing any popovers is unnecessary. From our perspective, it's kind of like the drag gesture is not present at all.

We aim to make the issue disappear with minimal, reasonable changes (unless a non-minimal solution is cleaner / better / more maintainable).

Reply:
> it may be seen as inconsistency and considered to be a regression from this PR.

Hm, I understand. I'll re-test to ensure we're on the same page here.

Reply:
> unless we are not aiming to closepopover when file is dragover on LHN or Popover itself

We are not. I gave this a thought, and if this is the only reason to include this block, I insist of removing this. After this gets a C+ approval (i.e. mine), you'll have a chance to add a note about this in the PR, and Joel will take this into consideration when giving his final review. 

> but considering **we do close the popover on RHP** we should also close on LHN for sake of consistency

What do you mean by this? Would you post a video of when we "close the popover on RHP" on `main`?

Reply:
Please...
- either provide a video of the undesired (from your perspective) behavior which happens when we delete this block
  - ...I will then decide whether I agree that the behavior on the video is indeed undesired
- ...or remove this block

Reply:
I tried to remove this myself and reproduce the issue with the RHP and wasn't able to.

Reply:
If you agree that it solves our original problem well and addresses the RHP-related concern, feel free to cherry-pick this commit.